providers/stores/mongo.js

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+const MongoClient = require('mongodb').MongoClient
+const promiseRetry = require('promise-retry')
+const EntityCoordinates = require('../../lib/entityCoordinates')
+
+class MongoStore {
+  constructor(options) {
+    this.options = options
+  }
+
+  initialize() {
+    return promiseRetry(async retry => {
+      try {
+        this.client = await MongoClient.connect(
+          this.options.connectionString,
+          { useNewUrlParser: true }
+        )
+        this.db = this.client.db(this.options.dbName)
+        this.collection = this.db.collection('definitions')
+      } catch (error) {
+        retry(error)
+      }
+    })
+  }
+
+  /**
+   * List all of the matching components for the given coordinates.
+   * Accepts partial coordinates.
+   *
+   * @param {EntityCoordinates} coordinates
+   * @returns A list of matching coordinates i.e. [ 'npm/npmjs/-/JSONStream/1.3.3' ]
+   */
+  async list(coordinates) {
+    const list = await this.collection.find({ id: this._getId(coordinates) }, 'id')
+    return list.map(entry => entry.id)
+  }
+
+  /**
+   * Get and return the object at the given coordinates.
+   *
+   * @param {Coordinates} coordinates - The coordinates of the object to get
+   * @returns The loaded object
+   */
+  get(coordinates) {
+    return this.collection.findOne({ id: this._getId(coordinates) }, { projection: { _id: 0, id: 0 } })
+  }
+
+  store(definition) {
+    definition.id = this._getId(definition.coordinates)
+    return this.collection.replaceOne({ id: definition.id }, definition, { upsert: true })
```

Comment:
this is actually leaking the `id` property back to the definition response.

When we call `get()` we project-away the id property

Reply:
Yep makes sense. We should just return the original value that is passed in